### PR TITLE
Fix null-handling of IN with constant in-list

### PR DESCRIPTION
**Pull request by kagamiori**
Summary: The ComplexTypeInPredicate class for constant in-list doesn't match Presto behavior on handling nested NULL elements (https://github.com/facebookincubator/velox/issues/10570). This diff fixes it.

Differential Revision: D60768330

### CODE REVIEW

File: velox/functions/prestosql/tests/InPredicateTest.cpp [hunk 1]
```diff
@@ -968,7 +970,7 @@ TEST_F(InPredicateTest, arrays) {
 
   auto inValuesWithNulls = makeNullableArrayVector<int32_t>({
       {1},
-      {1, std::nullopt, 2},
+      {1, std::nullopt},
```
Comment (kevinwilfong) [inline velox/functions/prestosql/tests/InPredicateTest.cpp:973]: Out of curiosity, why did you remove the existing value?  It seems like it would cover what you're trying to test (both sides contain null and are a match), why not leave it as is or add this as an additional value?

**Review by kevinwilfong — APPROVED** [review]
Thanks Wei!

File: velox/functions/prestosql/tests/InPredicateTest.cpp (hunk 1 above)
Comment (kagamiori) [reply velox/functions/prestosql/tests/InPredicateTest.cpp:973]: @kevinwilfong Good point. I will restore the {1, std::nullopt, 2} value and add an additional {1, 2, std::nullopt}.
